src/components/application_manager/test/commands/mobile/get_way_points_test.cc

```diff
@@ -0,0 +1,196 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "application_manager/commands/command_request_test.h"
+
+#include "mobile/get_way_points_request.h"
+
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+using namespace application_manager;
+
+using ::testing::Return;
+using ::testing::_;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 3u;
+const uint32_t kConnectionKey = kAppId;
+const std::string kMethodName = "Navigation.GetWayPoints";
+}
+
+class GetWayPointsRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetUp() OVERRIDE {
+    message_ = utils::MakeShared<SmartObject>(::smart_objects::SmartType_Map);
+    (*message_)[strings::msg_params] =
+        ::smart_objects::SmartObject(::smart_objects::SmartType_Map);
+
+    command_sptr_ =
+        CreateCommand<application_manager::commands::GetWayPointsRequest>(
+            CommandsTest::kDefaultTimeout_, message_);
+  }
+
+  MessageSharedPtr message_;
+  utils::SharedPtr<commands::GetWayPointsRequest> command_sptr_;
+};
+
+TEST_F(GetWayPointsRequestTest,
+       Run_InvalidApp_ApplicationNotRegisteredResponce) {
+  const int32_t kConnectionKey = 1;
+
+  (*message_)[strings::params][strings::connection_key] = kConnectionKey;
+
+  utils::SharedPtr<Application> null_application_sptr;
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(null_application_sptr));
+
+  CallRun caller(*command_sptr_);
+
+  MessageSharedPtr result_message = CatchMobileCommandResult(caller);
+
+  const mobile_api::Result::eType kResult =
+      static_cast<mobile_api::Result::eType>(
+          (*result_message)[strings::msg_params][strings::result_code].asInt());
+
+  EXPECT_EQ(mobile_api::Result::APPLICATION_NOT_REGISTERED, kResult);
+}
+
+TEST_F(GetWayPointsRequestTest, Run_ApplicationRegistered_Success) {
+  (*message_)[strings::params][strings::connection_key] = kConnectionKey;
+
+  MockAppPtr application_sptr = CreateMockApp();
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(application_sptr));
+  EXPECT_CALL(*application_sptr, app_id()).WillOnce(Return(1));
+
+  EXPECT_CALL(mock_app_manager_, GetNextHMICorrelationID())
+      .WillOnce(Return(kCorrelationId));
+
+  CallRun caller(*command_sptr_);
+
+  MessageSharedPtr result_message = CatchHMICommandResult(caller);
+
+  const hmi_apis::FunctionID::eType kResultFunctionId =
```

Comment:
@LevchenkoS , this tests has been written before updating rules. In this file i checked and fixed it. But in another it will be fixed with time waterfall :) 
Fixed in 	e17c66c